### ISSUE TITLE
Use CameraX

### ISSUE THREAD

**Issue by renanferrari** [firebase/mlkit-material-android#2]
Google recently announced CameraX, a Jetpack support library that wraps `camera2` capabilities into a much simpler API, based on use cases.

One of those use cases is _Image Analysis_, which seems perfect to use with MLKit – they even [mention it in their documentation](https://developer.android.com/training/camerax#primary) as an example.

I'd love to see this app implemented with CameraX. Is this something that would interest you?

**Comment by kenz** [comment]
So do I.
In my option, Camera X is more useful and easy for almost developers.
One problem about that Camera X is not support API Level 19(Kitkat).

**Comment by zhouyiself** [comment]
@renanferrari Yes, it's do in our plan but the priority may be not that high for now. Some factors we need to take into consideration before doing that,  e.g.
* CameraX is still in alpha stage and the minimum supported API is 21
* MLKit itself may also require some tweaks to work more nicely with it.

**Comment by spacemx09** [comment]
I have been trying to port the implementation with CameraX, however, I haven't been able to yet. The question I have is will it not be a problem if we keep using the deprecated Camera API? Our app uses minimum API 21, and I think most of the apps in market are following the same, as there are very less devices below 21 now a days. So please suggest which one would be the best to follow this implementation.

Update: Right after I posted, I have been able to get some success with CameraX. Took a reference from [Crysxd](https://github.com/crysxd/Object-Tracking-Demo/)'s implementation. However, still modifying for multi object detection.

**Comment by KaidoNinja** [comment]
Would really love to see this example with CameraX, any progress?

**Comment by spacemx09** [comment]
I have been successfully able to implement multiple object detection using CameraX with the help of [crysxd/Object-Tracking-Demo](https://github.com/crysxd/Object-Tracking-Demo/) and this demo. I will make it available once I get a little time. Also, we cannot put it in production as CameraX is in alpha. So I would suggest to try Camera2 if you need the feature to be in production right away.

I am still not sure if it is still okay to use Camera API as it is deprecated. Though it works fine on all the API upto the latest Q.

**Comment by lakshaymudgal** [comment]
Anybody here to me that how to put the captured photo in any view like (imageview, textureview, etc) in android.

**Comment by seventhmoon** [comment]
Since `android.hardware.Camera` was deprecated in API level 21.

I think the project should use [Camera2](https://developer.android.com/reference/android/hardware/camera2/package-summary) or CameraX if it is out of beta. (both require API level 21) 

Agree with @kenz, we should move forward.

**Comment by nelson336** [comment]
estou trabalhando com um aplicativo que min-sdk é 19 e não posso mudar. Não tem problema o camera x, ser utilizado apenas na 21. Mas podia deixar eu importar no projeto e limitar com if (Build.Version.Os). E não no build.

**Comment by nelson336** [comment]
I am working with an application that min-sdk is 19 and I cannot change. There is no problem with camera x, to be used only in 21. But I could import it into the project and limit it with if (Build.VERSION.SDK_INT). And not in the build.

**Comment by nelson336** [comment]
tools:overrideLibrary="androidx.camera.core" to force usage ??

**Comment by nelson336** [comment]
That if I can import camera x on android, I can use camera 1 for sdk less than 21 and for larger use camera x.
